modules/run-consul/run-consul

```diff
@@ -315,37 +316,45 @@ EOF
   chown "$user:$user" "$config_path"
 }
 
-function generate_supervisor_config {
-  local -r supervisor_config_path="$1"
+function generate_systemd_config {
+  local -r systemd_config_path="$1"
   local -r consul_config_dir="$2"
   local -r consul_data_dir="$3"
-  local -r consul_log_dir="$4"
-  local -r consul_bin_dir="$5"
-  local -r consul_user="$6"
-  shift 6
+  local -r consul_bin_dir="$4"
+  local -r consul_user="$5"
+  shift 5
   local -r environment=("$@")
-
-  log_info "Creating Supervisor config file to run Consul in $supervisor_config_path"
-  cat > "$supervisor_config_path" <<EOF
-[program:consul]
-command=$consul_bin_dir/consul agent -config-dir $consul_config_dir -data-dir $consul_data_dir
-stdout_logfile=$consul_log_dir/consul-stdout.log
-stderr_logfile=$consul_log_dir/consul-error.log
-numprocs=1
-autostart=true
-autorestart=unexpected
-exitcodes=0
-stopsignal=INT
-user=$consul_user
-environment=$(join_by "," "${environment[@]}")
+  local -r config_path="$consul_config_dir/$CONSUL_CONFIG_FILE"
+
+  log_info "Creating systemd config file to run Consul in $systemd_config_path"
+  cat > "$systemd_config_path" <<EOF
+[Unit]
+Description="HashiCorp Consul - A service mesh solution"
+Documentation=https://www.consul.io/
+Requires=network-online.target
+After=network-online.target
+ConditionFileNotEmpty=$config_path
```

Comment:
Same question here about stdout/stderr logs. If we want users to use `journalctl`, we should document how to do so.

Reply:
Ah yes, good point.  I've added some documentation about `journalctl` to view the logs, and also made the `StandardOutput` and `StandardError` config options available in case anyone wants to use them.